app/controllers/guide/scenarios_controller.rb

```diff
@@ -4,6 +4,9 @@ class Guide::ScenariosController < Guide::BaseController
   def show
     expose_layout
     expose_scenario
+  rescue NoMethodError => e
```

Comment:
I had a look at this code and i'm having a hard time following it. I'm guessing this method call is dynamic, can you link me to where it is? My concern with simply catching this error is if it is infact dyanmically calling methods from information in the URL it could be possible to craft a particular URL to have ruby run whatever code you want.

Is that the case, does it always simply call the `scenarios` method?

Reply:
Despite my recent contributions (for rails 5 compatibility), I'm not familiar with the internals of this code. 

Judging from the error and backtrace:

> NoMethodError: undefined method `scenarios' for #<Guide::Content::Structures::User::Downloads:0x007f9aa81d98f0>

Could we catch the error earlier, eg:

```ruby
  # app/models/guide/simulator.rb
  def scenario(scenario_id)
    # Add something like this and rescue in controller
    raise InvalidStructureError unless @structure.respond_to?(:scenarios)controller
    @structure.scenarios[scenario_id].tap do |scenario|
      raise Guide::Errors::InvalidScenario unless scenario.present?
    end
  end
```